ddtrace/internal/runtime/lightstep_metrics.py

```diff
@@ -0,0 +1,248 @@
+        net_recv = net_recv_total - self.stored_value["NET_RECV_TOTAL"]
+        net_sent = net_sent_total - self.stored_value["NET_SENT_TOTAL"]
+
+        system_memory = self.mem()
+
+        self.previous_value = self.stored_value
+
+        self.stored_value = dict(
+            CPU_TIME_SYS_TOTAL=cpu_time_sys_total,
+            CPU_TIME_USER_TOTAL=cpu_time_user_total,
+            SYSTEM_CPU_SYS_TOTAL=system_cpu_sys_total,
+            SYSTEM_CPU_USER_TOTAL=system_cpu_user_total,
+            SYSTEM_CPU_TOTAL=system_cpu_total_total,
+            SYSTEM_CPU_USAGE_TOTAL=system_cpu_usage_total,
+            NET_RECV_TOTAL=net_recv_total,
+            NET_SENT_TOTAL=net_sent_total,
+        )
+
+        return [
+            # process metrics
+            (LS_PROCESS_CPU_TIME_SYS, cpu_time_sys, MetricKind.COUNTER),
+            (LS_PROCESS_CPU_TIME_USER, cpu_time_user, MetricKind.COUNTER),
+            (LS_PROCESS_MEM_RSS, self.proc.memory_info().rss, MetricKind.GAUGE),
+            # system CPU metrics
+            (LS_SYSTEM_CPU_TIME_SYS, system_cpu_sys, MetricKind.COUNTER),
+            (LS_SYSTEM_CPU_TIME_USER, system_cpu_user, MetricKind.COUNTER),
+            (LS_SYSTEM_CPU_TIME_TOTAL, system_cpu_total, MetricKind.COUNTER),
+            (LS_SYSTEM_CPU_TIME_USAGE, system_cpu_usage, MetricKind.COUNTER),
+            # system memory metrics
+            (LS_SYSTEM_MEM_AVAIL, system_memory.available, MetricKind.GAUGE),
+            (LS_SYSTEM_MEM_USED, system_memory.used, MetricKind.GAUGE),
+            # system network metrics
+            (LS_SYSTEM_NET_RECV, net_recv, MetricKind.COUNTER),
+            (LS_SYSTEM_NET_SENT, net_sent, MetricKind.COUNTER),
+        ]
+
+    def collect_fn(self, keys):
+        with self.proc.oneshot():
+            metrics = self._measure()
+            if not self._skipped_first:
+                # intentionally skip the initial set of metrics
+                self._skipped_first = True
+                metrics = self._measure()
+            return metrics
+
+    def rollback(self):
+        self.stored_value = self.previous_value
+
+
+class LightstepRuntimeMetrics(RuntimeCollectorsIterable):
+    ENABLED = GC_RUNTIME_METRICS | LS_RUNTIME_METRICS
+    COLLECTORS = [
+        GCRuntimeMetricCollector,
+        LightstepPSUtilRuntimeMetricCollector,
+    ]
+
+    def rollback(self):
+        for c in self._collectors:
+            if hasattr(c, "rollback"):
+                c.rollback()
+
+
+class LightstepMetricsWorker(_worker.PeriodicWorkerThread):
+    """ Worker thread to collect and write metrics to a Lightstep endpoint """
+
+    _flush_interval = 30
+    _key_length = 30
+
+    def __init__(self, client, flush_interval=_flush_interval, tracer_tags={}):
+        super(LightstepMetricsWorker, self).__init__(interval=flush_interval, name=self.__class__.__name__)
+        self._component_name = tracer_tags.get("lightstep.component_name")
+        self._service_version = tracer_tags.get("service.version")
```

Comment:
Just a (probably wrong) hunch, should this be `lightstep.service_version`?

Reply:
it's correct as `service.version` but i've updated the code to use the constants instead of strings here